lang/macro/src/codegen/storage.rs

```diff
@@ -54,13 +54,6 @@ impl GenerateCode for Storage<'_> {
         let message_impls = self.generate_message_impls();
         let storage_struct = self.generate_storage_struct();
 
-        let use_emit_event = if !self.contract.events.is_empty() {
-            // Required to allow for `self.env().emit_event(..)` in messages and constructors.
-            quote! { use __ink_private::EmitEvent as _; }
-        } else {
-            quote! {}
-        };
-
```

Comment:
How can we do without this?

Reply:
Define an empty `Event` enum when no events are defined, rather than no enum at all?

Reply:
Heh sorry I realise I misunderstood your question originally. The answer is I don't know - it just worked. The trait must be in scope somehow. Anyway I have restored this.